1. Data Description


In [1]:
library(tidyverse)

players<-read_csv("data/players.csv")
sessions<-read_csv("data/sessions.csv")


── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimite

In [6]:
glimpse(players)
glimpse(sessions)

summary(players)
summary(sessions)


Rows: 196
Columns: 7
$ experience   <chr> "Pro", "Veteran", "Veteran", "Amateur", "Regular", "Amate…
$ subscribe    <lgl> TRUE, TRUE, FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, TRUE, T…
$ hashedEmail  <chr> "f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8…
$ played_hours <dbl> 30.3, 3.8, 0.0, 0.7, 0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 1.6, 0…
$ name         <chr> "Morgan", "Christian", "Blake", "Flora", "Kylie", "Adrian…
$ gender       <chr> "Male", "Male", "Male", "Female", "Male", "Female", "Fema…
$ Age          <dbl> 9, 17, 17, 21, 21, 17, 19, 21, 47, 22, 23, 17, 25, 22, 17…
Rows: 1,535
Columns: 5
$ hashedEmail         <chr> "bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8a…
$ start_time          <chr> "30/06/2024 18:12", "17/06/2024 23:33", "25/07/202…
$ end_time            <chr> "30/06/2024 18:24", "17/06/2024 23:46", "25/07/202…
$ original_start_time <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+12…
$ original_end_time   <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          


## Players dataset

- Number of rows (observations): **196**  
- Number of columns (variables): **7**

| Variable name | Type of variable | Variable meaning |
|----------------|------------------|------------------|
| experience     | character        | The player's skill level |
| subscribe      | logical          | Whether the player is subscribed to a game-related newsletter |
| hashedEmail    | character        | ASK TAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA |
| played_hours   | double           | Number of hours the player has played |
| name           | character        | Player's name |
| gender         | character        | Player's gender |
| age            | double           | Player's age |

#### Summary statistics 

| Variable | Min | 1st Qu. | Median | Mean | 3rd Qu. | Max | NA's |
|-----------|------|----------|---------|------|----------|------|----------|
| played_hours | 0.00 | 0.00 | 0.10 | 5.85 | 0.60 | 223.10 | 0 |
| Age | 9.00 | 17.00 | 19.00 | 21.14 | 22.75 | 58.00 | 2 |

#### Issues 

- 2 missing values in the column Age. 
- Some values in the column played_hours are 0.00, meaning they may have IDKKK


## Sessions dataset
- Number of rows (observations): **1,535**  
- Number of columns (variables): **5**

| Variable name | Type of variable | Variable meaning |
|----------------|------------------|------------------|
| hashedEmail    | character        | ASK TAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA |
| start_time     | character          | The time the session started |
| end_time       | character        | The time the session ended |
| original_start_time | double           | SK TAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA |
| original_end_time   | double        | SK TAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA |

#### Summary statistics

| Variable | Min | 1st Qu. | Median | Mean | 3rd Qu. | Max | NA's |
|-----------|------|----------|---------|------|----------|------|----------|
| original_start_time | 1.712×10¹² | 1.716×10¹² | 1.719×10¹² | 1.719×10¹² | 1.722×10¹² | 1.727×10¹² | 0 |
| original_end_time | 1.712×10¹² | 1.716×10¹² | 1.719×10¹² | 1.719×10¹² | 1.722×10¹² | 1.727×10¹² | 2 |

#### Issues

- 2 missing values in the column original_end_time.
- The columns original_start_time and original_end_time are difficult to interpret. 


any issues you see in the data, any other potential issues related to things you cannot directly see, how the data were collected, etc


